melodies_monet/driver.py

```diff
@@ -1509,7 +1514,24 @@ def plotting(self):
 
                         # Query selected points if applicable
                         if domain_type != 'all':
-                            pairdf_all.query(domain_type + ' == ' + '"' + domain_name + '"', inplace=True)
+                            if domain_query:
+                                pairdf_all.query(domain_type + ' == ' + '"' + domain_name + '"', inplace=True)
+                            else: 
+                                if domain_type == 'epa_region':
+                                    bounds = get_epa_region_bounds(acronym=domain_name)
+                                elif domain_type == 'giorgi_region':
+                                    bounds = get_giorgi_region_bounds(acronym=domain_name)
+                                else:
+                                    raise ValueError("Currently, region selections whithout a domain query have only"
+                                                     + "been implemented for Giorgi and EPA regions. You asked for"
+                                                     + f" {domain_type}. Soon, arbitrary rectangular boxes, US states and"
+                                                     + "others will be included.")
+                                pairdf = pairdf_all.loc[
+                                             (pairdf_all["latitude"] > bounds[0])
```

Comment:
Only works if region doesn't cross antimeridian and doesn't include N or S pole. And presumes bounds are in correct order and in [-180, 180) longitude format like the data. Fine for now but we might want to increase robustness in the future.

Reply:
Agreed. Since this is only implemented for Giorgi and EPA rectangles, it's not a problem for now. I suggest adding regionmask (check #272 ). This would, however, also require geopandas, since it's required by regionmask. I think that having it as an optional dependency would make sense.

Reply:
Do you think we should add that check?

Reply:
@rschwant @zmoon do we need anything else? I would like to have this solved before doing new PR's, to test the new code with this while avoiding branch conflicts.